bitnami/contour/values.yaml

```diff
@@ -259,9 +259,13 @@ contour:
   ## @param contour.extraEnvVarsSecret Secret containing extra env vars to be added to all Contour containers
   ##
   extraEnvVarsSecret: ""
-  ## @param contour.ingressClass Name of the ingress class to route through this controller
-  ##
-  ingressClass: contour
+  ## @param contour.ingressClass.name Name of the ingress class to route through this controller
+  ## @param contour.ingressClass.create Whether to create or not the IngressClass resource
+  ## @param contour.ingressClass.default Mark IngressClass resource as default for cluster
+  ingressClass:
+    name: contour
+    create: true
+    default: true
```

Comment:
This change introduce a backwards incompatibility since you're modifying the **values.yaml** structure. Therefore, a user trying to upgrade reusing a previous **values.yaml** that was setting the `contour.ingressClass` will find some errors.
There's a way to make the changes compatible which consists on playing with the `kindIs`/`typeIs` functions and adapt the manifests depending on the parameter type (string vs object)

Reply:
Thanks for your review and sorry for late response 🙏 
I added type checking on Deployment/IngressClass manifest, but I'm not sure about how to document multiple type support.

Temporarily, I add comment to `values.yaml` like below: https://github.com/bitnami/charts/blob/422dddf24a4883d7e7030651969535028166852c/bitnami/contour/values.yaml#L269-L271

WDYT?

Reply:
I'd add a warning so users understand this "string" approach is deprecated. Also, by our conventions all the comments describing a parameter be placed BEFORE the parameter. Sth like this:
```
  ## @param contour.ingressClass.name Name of the ingress class to route through this controller
  ## @param contour.ingressClass.create Whether to create or not the IngressClass resource
  ## @param contour.ingressClass.default Mark IngressClass resource as default for cluster
  ##
  ## DEPRECATED: Use a map instead
  ## You can use the the 'contour.ingressClass' as a string to indicate the ingress
  ## class name. This will skip the creation of an IngressClass resource
  ## e.g:
  ## ingressClass: contour
  ##
  ingressClass:
    name: contour
    create: true
    default: true
```